### PR TITLE
Filter panel

### LINKED ISSUE #268: Filter panel
Filter people by:

1. Group
2. Supervisor
3. Role
4. Service

### PR DESCRIPTION
**Pull request by nyet-ty**
Resolve #268

### CODE REVIEW

File: src/components/UsersPage/UsersPage.tsx [hunk 1]
```diff
@@ -1,24 +1,42 @@
...
+                filterState={filtersQuery}
+                total={usersQuery.data?.total || 0}
+                counter={usersQuery.data?.counter || 0}
+                setSearchQuery={setSearchQuery}
+                onFilterApply={setFiltersQuery}
+            />
+            <PageContent>
+                {usersQuery.data?.users.map((user) => (
+                    <div key={user.id}>
+                        <Link onClick={() => showUserPreview(user.id)} href={pages.user(user.id)}>
```
Comment (tsumo) [inline src/components/UsersPage/UsersPage.tsx:34]: Lets use UserListItem here

File: src/modules/groupSchemas.ts [hunk 2]
```diff
@@ -29,3 +29,11 @@ export const getGroupListSchema = z.object({
         .optional(),
 });
 export type GetGroupList = z.infer<typeof getGroupListSchema>;
+
+export const suggestionsGroupSchema = z.object({
```
Comment (tsumo) [inline src/modules/groupSchemas.ts:33]: getGroupSuggestionsSchema

File: src/modules/groupMethods.ts [hunk 3]
```diff
@@ -158,4 +159,20 @@ export const groupMethods = {
         }
         return { adjacencyList, dict };
     },
+    suggestions: async ({ query, include, take = suggestionsTake }: SuggestionsGroup) => {
```
Comment (tsumo) [inline src/modules/groupMethods.ts:162]: getSuggestions
also empty line between methods

File: src/modules/roleSchemas.ts [hunk 4]
```diff
@@ -27,3 +27,11 @@ export const getRoleListSchema = z.object({
         .optional(),
 });
 export type GetRoleList = z.infer<typeof getRoleListSchema>;
+
+export const suggestionsRoleSchema = z.object({
```
Comment (tsumo) [inline src/modules/roleSchemas.ts:31]: getRoleSuggestionsSchema

File: src/modules/roleMethods.ts [hunk 5]
```diff
@@ -38,4 +40,20 @@ export const roleMethods = {
             take: data.take,
         });
     },
+
+    suggestions: async ({ query, include, take = suggestionsTake }: SuggestionsRole) => {
```
Comment (tsumo) [inline src/modules/roleMethods.ts:44]: getSuggestions

File: src/modules/userSchemas.ts [hunk 6]
```diff
@@ -48,3 +51,11 @@ export const editUserSettingsSchema = z.object({
     theme: z.enum(themes).optional(),
 });
 export type EditUserSettings = z.infer<typeof editUserSettingsSchema>;
+
+export const suggestionsUserSchema = z.object({
```
Comment (tsumo) [inline src/modules/userSchemas.ts:55]: Same here

File: src/modules/userMethods.ts [hunk 7]
```diff
@@ -125,4 +159,24 @@ export const userMethods = {
             data: { name: data.name, supervisorId: data.supervisorId },
         });
     },
+    suggestions: async ({ query, include, take = suggestionsTake }: SuggestionsUser) => {
```
Comment (tsumo) [inline src/modules/userMethods.ts:162]: getSuggestions
empty line between methods

File: src/trpc/router/groupRouter.ts [hunk 8]
```diff
@@ -57,4 +63,8 @@ export const groupRouter = router({
     getHierarchy: protectedProcedure.input(z.string()).query(({ input }) => {
         return groupMethods.getHierarchy(input);
     }),
+
+    suggestions: protectedProcedure.input(suggestionsGroupSchema).query(({ input }) => {
```
Comment (tsumo) [inline src/trpc/router/groupRouter.ts:67]: getSuggestions

**Review by tsumo — CHANGES_REQUESTED** [review]

**Comment by tsumo** [comment]
Minor nitpicks, overall good job!
Also rebase is needed

File: src/components/LayoutMain.tsx [hunk 9]
```diff
@@ -15,6 +15,10 @@ const StyledContent = styled.div`
     min-height: calc(100vh - 160px);
 `;
 
+export const PageContent = styled.div`
+    padding: 10px 40px 0 40px;
```
[the remark this answers is not in the document]
Comment (nyet-ty) [reply src/components/LayoutMain.tsx:19]: I took PageContend from Issues, it in px where also, and gaps in rem, so I don't know how to use bricks gaps correctly

File: src/modules/roleMethods.ts (hunk 5 above)
Comment (nyet-ty) [reply src/modules/roleMethods.ts:44]: same method in issues named suggestions, so I'd like to keep it name

File: src/components/UsersPage/UsersPage.tsx (hunk 1 above)
Comment (nyet-ty) [reply src/components/UsersPage/UsersPage.tsx:34]: I don't think UserListItem suits here, I think it's better to make a new component in next commit

File: src/components/LayoutMain.tsx (hunk 9 above)
[the remark this answers is not in the document]
Comment (tsumo) [reply src/components/LayoutMain.tsx:19]: `padding: ${gapM} ${gapL} ${gapL} ${gapL}` should work on all our pages and will be consistent with the header paddings.

File: src/components/UsersPage/UsersPage.tsx (hunk 1 above)
Comment (tsumo) [reply src/components/UsersPage/UsersPage.tsx:34]: ok

File: src/modules/groupSchemas.ts [hunk 10]
```diff
@@ -29,3 +29,11 @@ export const getGroupListSchema = z.object({
...
 });
 export type GetGroupList = z.infer<typeof getGroupListSchema>;
+
+export const getGroupSuggestionsSchema = z.object({
+    query: z.string(),
+    take: z.number().optional(),
+    include: z.array(z.string()).optional(),
+});
+
+export type SuggestionsGroup = z.infer<typeof getGroupSuggestionsSchema>;
```
Comment (tsumo) [inline src/modules/groupSchemas.ts:39]: `GetGroupSuggestions`, for consistency with the schema naming

File: src/modules/roleMethods.ts (hunk 5 above)
Comment (tsumo) [reply src/modules/roleMethods.ts:44]: ok

File: src/modules/groupSchemas.ts (hunk 10 above)
Comment (nyet-ty) [reply src/modules/groupSchemas.ts:39]: missed that

**Review by tsumo — APPROVED** [review]

**Review by 9teen90nine — APPROVED** [review]

**Pull request merged**